### PR TITLE
repo setup / auto-release setup

### PR DESCRIPTION
**Pull request by mwear**
This PR includes adds a changelog, code of conduct and contributing documents. It also adds a tag based auto-release process based on our other JS libraries.

### CODE REVIEW

**Comment by mwear** [comment]
cc: @obecny

File: scripts/release-prepare.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  echo "$checkNpm"
+  echo "Please fix the error and try again"
+  exit
+fi
+
+upstreamExists=`git remote -v | grep upstream_tmp_release | tail -1`
+if [ "$upstreamExists" != "" ]; then
+  git remote remove upstream_tmp_release
+fi
+git remote add upstream_tmp_release git@github.com:lightstep/opentelemetry-exporter-js.git
```
Comment (obecny) [inline scripts/release-prepare.sh]: update url

**Review by obecny — COMMENTED** [review]
2 things
the url is wrong, 
make sure you add `NPM_TOKEN` in circleCI

File: scripts/release-prepare.sh (hunk 1 above)
Comment (mwear) [reply scripts/release-prepare.sh]: Nice catch. `NPM_TOKEN` has been added in circle.

**Review by obecny — APPROVED** [review]

**Pull request merged**